src/Strategy/Haskell/Stack.hs

```diff
@@ -0,0 +1,127 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Strategy.Haskell.Stack
+  ( discover,
+    -- * Testing
+    buildGraph,
+    PackageName (..),
+    StackDep (..),
+    StackLocationType (..),
+  )
+where
+
+import Control.Effect.Diagnostics
+import Data.Aeson.Types
+import Data.Foldable (for_, find)
+import qualified Data.Map.Strict as M
+import Data.Text (Text)
+import qualified Data.Text as T
+import Discovery.Walk
+import Effect.Exec
+import Effect.Grapher
+import qualified Graphing as G
+import Path
+import Types
+import Prelude
+import Control.Monad (when)
+
+newtype PackageName = PackageName {unPackageName :: Text} deriving (FromJSON, Eq, Ord, Show)
+newtype StackLocation = StackLocation {unStackLocation :: StackLocationType} deriving (Eq, Ord, Show)
+
+data StackDep = StackDep
+  { stackName :: PackageName,
+    stackVersion :: Text,
+    stackDepNames :: [PackageName],
+    stackLocation :: StackLocationType
+  }
+  deriving (Eq, Ord, Show)
+
+data StackLocationType
+  = Local
+  | Remote
+  | BuiltIn
+  deriving (Eq, Ord, Show)
+
+instance FromJSON StackDep where
+  parseJSON = withObject "StackDep" $ \obj ->
+    StackDep <$> obj .: "name"
+      <*> obj .: "version"
+      <*> obj .:? "dependencies" .!= []
+      <*> fmap unStackLocation (obj .:? "location" .!= StackLocation BuiltIn)
```

Comment:
rather than using a `StackLocation` wrapper around `StackLocationType`, directly define a `FromJSON` instance on `StackLocationType`
```suggestion
      <*> obj .:? "location" .!= BuiltIn
```